include/envoy/http/original_ip_detection.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+#include "envoy/http/codes.h"
+#include "envoy/http/header_map.h"
+#include "envoy/network/address.h"
+
+namespace Envoy {
+namespace Http {
+
+struct OriginalIPDetectionParams {
+  // Note that while extensions can modify the headers, they will undergo standard Envoy
```

Comment:
Silly but also document what these are
// The request headers from downstream.
// Note that ...